diff --git a/src/Plugin/Network/HidAuth.php b/src/Plugin/Network/HidAuth.php
index 49d1418..d54e65f 100644
--- a/src/Plugin/Network/HidAuth.php
+++ b/src/Plugin/Network/HidAuth.php
@@ -13,6 +13,7 @@ use Drupal\social_auth_hid\Settings\HidAuthSettings;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use League\OAuth2\Client\Provider\HumanitarianId;
 use Drupal\Core\Site\Settings;
+use Drupal\Core\Url;
 
 /**
  * Defines a Network Plugin for Social Auth Humanitarian ID.
@@ -140,7 +141,10 @@ class HidAuth extends NetworkBase implements HidAuthInterface {
 
     if ($this->validateConfig($settings)) {
       // All these settings are mandatory.
-      $redirect_url = \Drupal::request()->getSchemeAndHttpHost() . '/user/login/hid/callback';
+      $redirect_url = Url::fromUserInput('/user/login/hid/callback', [
+        'absolute' => TRUE,
+      ])->toString();
+
       $league_settings = [
         'clientId' => $settings->getClientId(),
         'clientSecret' => $settings->getClientSecret(),
