diff --git a/src/Plugin/QueueWorker/LinkStatusHandle.php b/src/Plugin/QueueWorker/LinkStatusHandle.php
index b869154..c7b3558 100644
--- a/src/Plugin/QueueWorker/LinkStatusHandle.php
+++ b/src/Plugin/QueueWorker/LinkStatusHandle.php
@@ -94,7 +94,9 @@ class LinkStatusHandle extends QueueWorkerBase implements ContainerFactoryPlugin
           continue;
         }
 
-        $handler->handle($link, $response);
+        if ($link) {
+          $handler->handle($link, $response);
+        }
       }
       catch (\Throwable $e) {
         // If we can`t load Link or entity - just skip it.
