diff --git a/src/Plugin/LinkStatusHandlerBase.php b/src/Plugin/LinkStatusHandlerBase.php
index 83c1894..2d2f03c 100644
--- a/src/Plugin/LinkStatusHandlerBase.php
+++ b/src/Plugin/LinkStatusHandlerBase.php
@@ -176,7 +176,14 @@ abstract class LinkStatusHandlerBase extends PluginBase implements LinkStatusHan
    */
   protected function switchSession() {
     // Switch anonymous user to an admin.
-    $this->accountSwitcher->switchTo(new UserSession(['uid' => user_load_by_name($this->linkcheckerSetting->get('error.impersonate_account'))]));
+    $user = user_load_by_name($this->linkcheckerSetting->get('error.impersonate_account'));
+    if ($user) {
+      $uid = $user->uid;
+    } else {
+      $uid = $this->linkcheckerSetting->get('error.impersonate_account');
+    }
+
+    $this->accountSwitcher->switchTo(new UserSession(['uid' => $uid]));
   }
 
   /**
