diff --git a/lib/Configuration.php b/lib/Configuration.php
index a969fb5..86404fa 100644
--- a/lib/Configuration.php
+++ b/lib/Configuration.php
@@ -63,6 +63,34 @@ class Configuration
     protected $tempFolderPath;
     protected $timeout = 120;

+    public $accountExport = NULL;
+    public $accountExports = NULL;
+    public $activityFeed = NULL;
+    public $authorizedApps = NULL;
+    public $automations = NULL;
+    public $batchWebhooks = NULL;
+    public $batches = NULL;
+    public $campaignFolders = NULL;
+    public $campaigns = NULL;
+    public $connectedSites = NULL;
+    public $conversations = NULL;
+    public $customerJourneys = NULL;
+    public $ecommerce = NULL;
+    public $facebookAds = NULL;
+    public $fileManager = NULL;
+    public $landingPages = NULL;
+    public $lists = NULL;
+    public $ping = NULL;
+    public $reporting = NULL;
+    public $reports = NULL;
+    public $root = NULL;
+    public $searchCampaigns = NULL;
+    public $searchMembers = NULL;
+    public $Surveys = NULL;
+    public $templateFolders = NULL;
+    public $templates = NULL;
+    public $verifiedDomains = NULL;
+
     public function __construct()
     {
         $this->tempFolderPath = sys_get_temp_dir();
