diff --git a/lib/Configuration.php b/lib/Configuration.php
index a969fb5..86404fa 100644
--- a/lib/Configuration.php
+++ b/lib/Configuration.php
@@ -63,6 +63,34 @@ class Configuration
     protected $tempFolderPath;
     protected $timeout = 120;
 
+    protected $accountExport = NULL;
+    protected $accountExports = NULL;
+    protected $activityFeed = NULL;
+    protected $authorizedApps = NULL;
+    protected $automations = NULL;
+    protected $batchWebhooks = NULL;
+    protected $batches = NULL;
+    protected $campaignFolders = NULL;
+    protected $campaigns = NULL;
+    protected $connectedSites = NULL;
+    protected $conversations = NULL;
+    protected $customerJourneys = NULL;
+    protected $ecommerce = NULL;
+    protected $facebookAds = NULL;
+    protected $fileManager = NULL;
+    protected $landingPages = NULL;
+    protected $lists = NULL;
+    protected $ping = NULL;
+    protected $reporting = NULL;
+    protected $reports = NULL;
+    protected $root = NULL;
+    protected $searchCampaigns = NULL;
+    protected $searchMembers = NULL;
+    protected $Surveys = NULL;
+    protected $templateFolders = NULL;
+    protected $templates = NULL;
+    protected $verifiedDomains = NULL;
+
     public function __construct()
     {
         $this->tempFolderPath = sys_get_temp_dir();
